### PR TITLE
Fix cwl and wdl dependency bleed and add stand alone tests.

### LINKED ISSUE #3565: toil does not install boto3 and boto in prepare step for toil-cwl-runner
After installing toil using:
```
make prepare
make develop extras=[cwl]
```
I get ModuleNotFoundErrors for boto and boto3 from toil-cwl-runner. Maybe boto should be added to cwl_reqs here: https://github.com/DataBiosphere/toil/blob/master/setup.py#L66

┆Issue is synchronized with this [Jira Task](https://ucsc-cgl.atlassian.net/browse/TOIL-878)
┆Issue Number: TOIL-878

### PR DESCRIPTION
**Pull request by DailyDreaming**
Resolves #3565.

## Changelog Entry
To be copied to the [draft changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog) by merger:

 * Fix cwl and wdl dependency bleed and add stand alone tests.

## Reviewer Checklist

 * [x] Make sure it is coming from `issues/XXXX-fix-the-thing` in the Toil repo, or from an external repo.
    * [ ] If it is coming from an external repo, make sure to pull it in for CI with:
        ```
        contrib/admin/test-pr otheruser theirbranchname issues/XXXX-fix-the-thing
        ```
    * [ ] If there is no associated issue, [create one](https://github.com/DataBiosphere/toil/issues/new).
* [x] Read through the code changes. Make sure that it doesn't have:
    * [x] Addition of trailing whitespace.
    * [x] New variable or member names in `camelCase` that want to be in `snake_case`.
    * [x] New functions without [type hints](https://docs.python.org/3/library/typing.html).
    * [x] New functions or classes without informative docstrings.
    * [x] Changes to semantics not reflected in the relevant docstrings.
    * [x] New or changed command line options for Toil workflows that are not reflected in `docs/running/cliOptions.rst`
    * [x] New features without tests.
* [x] Comment on the lines of code where problems exist with a review comment. You can shift-click the line numbers in the diff to select multiple lines.
* [ ] Finish the review with an overall description of your opinion.

## Merger Checklist

* [ ] Make sure the PR passes tests.
* [ ] Make sure the PR has been reviewed **since its last modification**. If not, review it.
* [ ] Merge with the Github "Squash and merge" feature.
    * [ ] If there are multiple authors' commits, add [Co-authored-by](https://github.blog/2018-01-29-commit-together-with-co-authors/) to give credit to all contributing authors.
* [ ] Copy its recommended changelog entry to the [Draft Changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog).
* [ ] Append the issue number in parentheses to the changelog entry.

### CODE REVIEW

File: src/toil/lib/aws/__init__.py [hunk 1]
```diff
@@ -25,3 +25,12 @@ def client(*args, **kwargs):
 def resource(*args, **kwargs):
     import boto3
     return boto3.resource(*args, **kwargs)
+
+
+def zone_to_region(zone: str):
```
Comment (adamnovak) [inline src/toil/lib/aws/__init__.py:30]: This isn't really a new function, but if we're touching it it couldn't hurt to give it a return type hint.

File: src/toil/lib/aws/__init__.py [hunk 2]
```diff
@@ -25,3 +25,12 @@ def client(*args, **kwargs):
...
+    """Get a region (e.g. us-west-2) from a zone (e.g. us-west-1c)."""
+    availability_zone = re.compile(r'^([a-z]{2}-[a-z]+-[1-9][0-9]*)([a-z])$')
```
Comment (adamnovak) [inline src/toil/lib/aws/__init__.py:32]: I think we had this outside the function so we wouldn't recompile it all the time. But maybe `re` memoizes these internally.

**Review by adamnovak — APPROVED** [review]
Looks good to me. There are a couple things I would have done differently, but this way is probably fine too.

File: src/toil/lib/aws/__init__.py (hunk 2 above)
Comment (DailyDreaming) [reply src/toil/lib/aws/__init__.py:33]: I believe it is: https://stackoverflow.com/questions/12514157/how-does-pythons-regex-pattern-caching-work

Comment (DailyDreaming) [reply src/toil/lib/aws/__init__.py:33]: I've added a comment.

File: src/toil/lib/aws/__init__.py (hunk 1 above)
Comment (DailyDreaming) [reply src/toil/lib/aws/__init__.py]: Fixed.

**Comment by DailyDreaming** [comment]
@adamnovak Cool, thanks for the review.  I believe I've addressed your comments.  Just have to wait for tests to pass now.

**Pull request merged**